module: document `parentURL` in register options (#51039)
nodejs/node: doc/api/module.md

  URL, such as `import.meta.url`, you can pass that URL here. **Default:**
  `'data:'`
* `options` {Object}
  * `parentURL` {string|URL} If you want to resolve `specifier` relative to a
    base URL, such as `import.meta.url`, you can pass that URL here. This

>     property is ignored if the `parentURL` is supplied in the second parameter.

Reviewer: ```suggestion
    property is ignored if the `parentURL` is supplied as the second argument.
```

  URL, such as `import.meta.url`, you can pass that URL here. **Default:**
  `'data:'`
* `options` {Object}
  * `parentURL` {string|URL} If you want to resolve `specifier` relative to a

>     base URL, such as `import.meta.url`, you can pass that URL here.

Reviewer: We should add which will be ignored if both are supplied
